### ISSUE TITLE
Array.Contains not working

### ISSUE THREAD

**Issue by davideicardi** [davideicardi/DynamicExpresso#68]
Array.Contains doesn't work.

		[Test]
		public void GitHub_Issue_()
		{
			var interpreter = new Interpreter();

			var array = new[] {5, 10, 6};

			interpreter.SetVariable("array", array);

			Assert.AreEqual(array.Contains(5), interpreter.Eval("array.Contains(5)"));
			Assert.AreEqual(array.Contains(3), interpreter.Eval("array.Contains(3)"));
		}

Se also https://stackoverflow.com/questions/46607171/parse-as-expression-an-array-containing-specific-identifier

**Comment by AbdoDabbas** [comment]
In file `Parser.cs` line 1380:
`method.MethodBase = methodInfo.MakeGenericMethod(genericArgsType.ToArray());`
This where the exception is thrown, I'm not an expert in C#, can you explain why are you trying to do this ?
Does this method `MakeGenericMethod` means like: "Try to get me back a method applicable to these parameters" ??

**Comment by AbdoDabbas** [comment]
Can't we change line 1380:
`method.MethodBase = methodInfo.MakeGenericMethod(genericArgsType.ToArray());`
To:
```
bool isExtension = methodInfo.IsDefined(typeof(ExtensionAttribute), true);
   if (isExtension)
       method.MethodBase = methodInfo.MakeGenericMethod(genericArgsType.Skip(1).ToArray());
   else
       method.MethodBase = methodInfo.MakeGenericMethod(genericArgsType.ToArray());
```
It looks like you're passing all arguments to the extension method, while should skip the first one I think.
Not sure or this solution and its effect on other methods but this what made the extension method work.

If it's correct let me know to make a pull request.

**Comment by davideicardi** [comment]
Maybe this can be a fix, but I should investigate.
If you can I suggest to try it and then run all unit tests, it should probably be a good proof that it is correct. If correct just fork the project, fix it and then create a pull request.

**Comment by xalikoutis** [comment]
Hi
Whats the status?

**Comment by davideicardi** [comment]
Fixed in v2.0.2
